### PR TITLE
expand admin[-access] support

### LINKED ISSUE #1629: Spawner.options_form is not shown when servers are started from the admin panel
which prevents servers from spawning, because the `user_options` is empty and throws `KeyError` when the spawner tries to get values from it.

**How to reproduce the issue**

1. Enable `admin_access` and add (an) admin user(s)
2. Add an `options_form` to spawner and use some field value from the form in the spawner by accessing `user_options`.
3. Log in as admin and go to admin panel
4. Push the `start server` button.

**What you expected to happen**

The form is shown.

**What actually happens**

An `API request failed (500): error` error raised, because an attempt to use the `user_options` in the Spawner ended with the `KeyError` exception.

**Share what version of JupyterHub you are using**

From the `jupyterhub/jupyterhub:latest` docker image.

### PR DESCRIPTION
**Pull request by minrk**
- [x] admin request for /user/:name spawns user's server, not redirecting to admin's server
- [x] admins can see/submit spawn form for users (if admin access is enabled)
- [x] spawn links in admin panel should trigger spawn form if defined

closes #1629

### CODE REVIEW

**Comment by consideRatio** [comment]
Wieee!!! @minrk this is a great improvement!

File: share/jupyterhub/static/js/admin.js [hunk 1]
```diff
@@ -121,7 +135,10 @@ require(["jquery", "bootstrap", "moment", "jhapi", "utils"], function ($, bs, mo
         });
     });
 
+<<<<<<< HEAD:share/jupyterhub/static/js/admin.js
```
Comment (willingc) [inline share/jupyterhub/static/js/admin.js]: Looks like some merge conflict cruft.

**Comment by minrk** [comment]
Rebased to fix the erroneous conflict marker

**Comment by willingc** [comment]
Thanks  @minrk.

**Pull request merged**